odd_collector/adapters/duckdb/client.py

```diff
@@ -0,0 +1,86 @@
+import os
+import re
+from duckdb import connect, DuckDBPyConnection
+
+
+class DuckDBClient:
+    def __init__(self, paths: list[str]):
+        self.db_files = self.__get_db_files(paths)
+
+    def __validate_paths(self, paths: list[str]):
```

Comment:
`paths` could be a list of Path types, that way you do not need to use `os` package